### PR TITLE
Adding Jupyter notebook preview support

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
This adds a lot of code, but we are taking most of the json data included in
the python notebook and putting it here. Also I'm including a lot of libraries
like react-markdown, dompurify and katex.

#### Ticket Link
#28

#### Screenshots
imagen

### CODE REVIEW

File: webapp/src/components/notebook_preview/notebook_preview.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    overflow: scroll;
+    max-width: 1024px;
+`;
+
+const NotebookPreview = ({fileInfo}: Props) => {
+    const [notebook, setNotebook] = useState<any>(null);
+
+    useEffect(() => {
+        const client = new Client4();
+        fetch(client.getFileUrl(fileInfo.id, new Date().getTime())).then(async (response: any) => {
```
Comment (crspeller) [inline webapp/src/components/notebook_preview/notebook_preview.tsx]: Isn't the second argument of getFileUrl supposed to be the update at of the fileinfo? We can also just leave it out to not break caching I think.

File: webapp/src/components/notebook_preview/notebook_preview.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            const data = await response.json();
+            setNotebook(data);
+        });
+    }, []);
```
Comment (crspeller) [inline webapp/src/components/notebook_preview/notebook_preview.tsx]: We should probably add `fileInfo.id` as a dependency here.

File: webapp/src/components/notebook_preview/code_cell_output_data.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+import 'katex/dist/katex.min.css';
+
+type Props = {
+    format: string
+    content: string|string[]
+}
+
+const CodeCellOutputData = ({format, content}: Props) => {
+    const sanitizedContent = DOMPurify.sanitize(joinText(content));
```
Comment (crspeller) [inline webapp/src/components/notebook_preview/code_cell_output_data.tsx:18]: @jupenur Do you have some time to make sure we are not missing an injection attack here?

**Review by crspeller — CHANGES_REQUESTED** [review]
This looks great! Tested it locally with some of my notebooks and seen no issue.

File: webapp/src/components/notebook_preview/notebook_preview.tsx (hunk 1 above)
Comment (jespino) [reply webapp/src/components/notebook_preview/notebook_preview.tsx]: Fixed

File: webapp/src/components/notebook_preview/notebook_preview.tsx (hunk 2 above)
Comment (jespino) [reply webapp/src/components/notebook_preview/notebook_preview.tsx]: Fixed

**Comment by crspeller** [comment]
Too many possible security issues for now. Closing as stale.

**Pull request closed without merging**